cetmix_tower_git/models/cx_tower_server.py

```diff
@@ -59,3 +59,40 @@ def _update_or_create_related_record(
         return super()._update_or_create_related_record(
             model, reference, values, create_immediately=create_immediately
         )
+
+    @api.model
+    def get_servers_by_git_ref(self, repository_url, head=None, head_type=None):
+        """
+        Find servers linked to a Git repo (URL must be pre-normalized).
+
+        Args:
+            repository_url (str): Normalized Git URL (use normalize_git_url() first)
+            head (str, optional): Branch/commit/PR identifier
+            head_type (str, optional): 'branch'|'commit'|'pr'
+
+        Returns:
+            Server recordset: Matching servers or empty if none found.
+        """
+        # URL MUST be already canonical.
+        if not repository_url:
+            return self.env["cx.tower.server"]
+
+        # Simplify 'refs/heads/main' to 'main' for easier comparison
+        head_simple = head.rsplit("/", 1)[-1] if head else False
```

Comment:
Once again: this function receives sanitised vals. No need to split to sanitise them again.